src/main/resources/public/js/activity.service.js

```diff
@@ -54,45 +54,46 @@
           addTimeForKm(activity, seconds, distanceHalfMarathon);
         }
         // calculate time for every distance starting with nearest km (round ceil)
-        for (var i = Math.ceil(activity.distance); i >= 1; i--) {
+        for (let index = Math.ceil(activity.distance); index >= 1; index -= 1) {
```

Comment:
That's a bit weird. Maybe we could at least use `allowForLoopAfterthoughts` from the first link. Because it should be safe within for-loops. What do you think?